src/settings_table.cpp

```diff
@@ -150,6 +150,12 @@ static void UpdateAllServiceInterval(int32_t new_value)
 		vds->servint_roadveh  = DEF_SERVINT_PERCENT;
 		vds->servint_aircraft = DEF_SERVINT_PERCENT;
 		vds->servint_ships    = DEF_SERVINT_PERCENT;
+	} else if (_game_mode == GM_MENU ? _settings_newgame.economy.use_realtime_units : _settings_game.economy.use_realtime_units) {
```

Comment:
Here too, I am a bit confused why these `_game_mode` checks exists. That would mean the title game can be influenced by the current settings? That makes no sense to me :)

So let's see if I can figure it out ...

If the game is in the main menu, you use the settings from `_settings_newgame`, which are the upcoming settings and can be changed freely in the settings menu. In that case, you use those settings in the .. title game?

Yeah, no, I can't make heads nor tails from this, sorry :) I need a bit of help understanding this :D

Reply:
Yes, in the main menu _settings_game refers to the title game, which is why in this mode we always work with _settings_newgame to change the newgame settings for the upcoming game.